File: cakk-api/src/main/java/com/cakk/api/service/user/UserService.java
package com.cakk.api.service.user;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import lombok.RequiredArgsConstructor;

import com.cakk.api.dto.response.user.ProfileInformationResponse;
import com.cakk.api.mapper.UserMapper;
import com.cakk.core.facade.user.UserManageFacade;
import com.cakk.core.facade.user.UserReadFacade;
import com.cakk.domain.mysql.dto.param.user.ProfileUpdateParam;
import com.cakk.domain.mysql.entity.user.User;
import com.cakk.domain.mysql.entity.user.UserWithdrawal;

@Service
@RequiredArgsConstructor
public class UserService {

	private final UserReadFacade userReadFacade;
	private final UserManageFacade userManageFacade;

	@Transactional(readOnly = true)
	public ProfileInformationResponse findProfile(final User signInUser) {
		final User user = userReadFacade.findByUserId(signInUser.getId());

		return UserMapper.supplyProfileInformationResponseBy(user);
	}

	@Transactional
	public void updateInformation(final ProfileUpdateParam param) {
		final User user = userReadFacade.findByUserId(param.userId());
		userManageFacade.updateProfile(user, param);
	}

	@Transactional
	public void withdraw(final User signInUser) {
		final User user = userReadFacade.findByIdWithAll(signInUser.getId());
		final UserWithdrawal withdrawal = UserMapper.supplyUserWithdrawalBy(user);

		userManageFacade.withdraw(user, withdrawal);
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/service/user/UserService.kt
package com.cakk.core.service.user

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

import com.cakk.core.dto.response.user.ProfileInformationResponse
import com.cakk.core.facade.user.UserManageFacade
import com.cakk.core.facade.user.UserReadFacade
import com.cakk.core.mapper.supplyProfileInformationResponseBy
import com.cakk.core.mapper.supplyUserWithdrawalBy
import com.cakk.infrastructure.persistence.entity.user.UserEntity

@Service
class UserService(
	private val userReadFacade: UserReadFacade,
	private val userManageFacade: UserManageFacade
) {


	@Transactional(readOnly = true)
	fun findProfile(signInUserEntity: UserEntity): ProfileInformationResponse {
		val user = userReadFacade.findByUserId(signInUserEntity.id)

		return supplyProfileInformationResponseBy(user)
	}

	@Transactional
	fun updateInformation(param: com.cakk.infrastructure.persistence.param.user.ProfileUpdateParam) {
		val user = userReadFacade.findByUserId(param.userId)
		userManageFacade.updateProfile(user, param)
	}

	@Transactional
	fun withdraw(signInUserEntity: UserEntity) {
		val user = userReadFacade.findByIdWithAll(signInUserEntity.id)
		val withdrawal = supplyUserWithdrawalBy(user)

		userManageFacade.withdraw(user, withdrawal)
	}
}
